src/DefaultWebFront.php

```
@@ -38,12 +49,81 @@ private function enableDebugLogging(MasterFactory $masterFactory, CommonFactory
         $masterFactory->register(new LoggingDomainEventHandlerFactory($commonFactory));
     }
 
-    protected function registerRouters(HttpRouterChain $routerChain)
+    final protected function getHandlerForRequest(HttpRequest $request): HttpRequestHandler
     {
-        $routerChain->register($this->getMasterFactory()->createProductSearchResultRouter());
-        $routerChain->register($this->getMasterFactory()->createProductDetailViewRouter());
-        $routerChain->register($this->getMasterFactory()->createProductListingRouter());
-        $routerChain->register($this->getMasterFactory()->createUnknownHttpRequestMethodRouter());
-        $routerChain->register($this->getMasterFactory()->createResourceNotFoundRouter());
+        if ($request instanceof HttpUnknownMethodRequest) {
+            return $this->getWebRequestHandlerLocator()->getRequestHandlerForCode(
+                UnknownHttpRequestMethodHandler::CODE,
+                ''
+            );
+        }
+
+        $metaJson = $this->getMetaJson($request);
+        $requestHandlerCode = $this->getRequestHandlerCode($request, $metaJson);
+
+        return $this->getWebRequestHandlerLocator()->getRequestHandlerForCode($requestHandlerCode, $metaJson);
+    }
+
+    private function getRequestHandlerCode(HttpRequest $request, string $metaInfoJson): string
+    {
+        $metaData = json_decode($metaInfoJson, true);
+
+        if (! isset($metaData[PageMetaInfoSnippetContent::KEY_HANDLER_CODE])) {
+            throw new MalformedMetaSnippetException('Request handler code is missing in meta snippet.');
```

Alternatively could return `ResourceNotFoundRequestHandler` but this smells with silent error to me.